### ISSUE TITLE
read clustering issue

### ISSUE THREAD

**Issue by mfranchett** [genomicsITER/NanoCLUST#54]
Hi, 

I keep getting errors while clustering reads, see error code below. Does anyone know how to troubleshoot this? 

Thanks in advance. 

#!/usr/bin/env python

  import numpy as np
  import umap
  import os
  os.environ['MPLCONFIGDIR'] = "~/NanoCLUST"
  import matplotlib.pyplot as plt
  from sklearn import decomposition
  import random
  import pandas as pd
  import hdbscan

  df = pd.read_csv("freqs.txt", delimiter="     ")

  #UMAP
  motifs = [x for x in df.columns.values if x not in ["read", "length"]]
  X = df.loc[:,motifs]
  X_embedded = umap.UMAP(n_neighbors=15, min_dist=0.1, verbose=2).fit_transform(X)

  df_umap = pd.DataFrame(X_embedded, columns=["D1", "D2"])
  umap_out = pd.concat([df["read"], df["length"], df_umap], axis=1)

  #HDBSCAN
  X = umap_out.loc[:,["D1", "D2"]]
  umap_out["bin_id"] = hdbscan.HDBSCAN(min_cluster_size=int(50), cluster_selection_epsilon=int(0.5)).fit_predict(X)

  #PLOT
  plt.figure(figsize=(20,20))
  plt.scatter(X_embedded[:, 0], X_embedded[:, 1], c=umap_out["bin_id"], cmap='Spectral', s=1)
  plt.xlabel("UMAP1", fontsize=18)
  plt.ylabel("UMAP2", fontsize=18)
  plt.gca().set_aspect('equal', 'datalim')
  plt.title("Projecting " + str(len(umap_out['bin_id'])) + " reads. " + str(len(umap_out['bin_id'].unique())) + " clusters generated by HDBSCAN", fontsize=18)

  for cluster in np.sort(umap_out['bin_id'].unique()):
      read = umap_out.loc[umap_out['bin_id'] == cluster].iloc[0]
      plt.annotate(str(cluster), (read['D1'], read['D2']), weight='bold', size=14)

  plt.savefig('hdbscan.output.png')
  umap_out.to_csv("hdbscan.output.tsv", sep="   ", index=False)

Command exit status:
  139

Command output:
  (empty)

Command error:
  .command.run: line 159:    23 Segmentation fault      (core dumped) /usr/bin/env python .command.sh

Work dir:
  /home/ec2-user/NanoCLUST/work/0c/e056c7f5bc18e9ca237da8653175f5

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

**Comment by liuweisu** [comment]
Hi! Have you solved the issue? It seems that I also have the same error when I try to run it on my own data.
